File: 27-RemoveElement/main.py
from typing import List

class Solution:
	def remove_elenment(self, nums: List[int], val: int) -> int:
		if len(nums) == 0:
			return 0

		i = 0
		for j in range(len(nums)):
			if nums[j] != val:
				nums[i] = nums[j]
				i += 1

		return i

s = Solution()
print(s.remove_elenment([3, 2, 2, 3], 3))
print(s.remove_elenment([0, 1, 2, 2, 3, 0, 4, 2], 2))

File: 27-RemoveElement/Main.java
class RemoveElement {
  public int removeElement(int[] nums, int val) {
    if (nums.length == 0) {
      return 0;
    }

    int i = 0;
    for (int j = 0; j < nums.length; j++) {
      if (nums[j] != val) {
        nums[i] = nums[j];
        i++;
      }
    }

    return i;
  }

  public static void main(String[] args) {
    RemoveElement sol = new RemoveElement();
    System.out.println(sol.removeElement(new int[] {3, 2, 2, 3}, 3));
    System.out.println(sol.removeElement(new int[] {0, 1, 2, 2, 3, 0, 4, 2}, 2))
  }
}